test/ts/collector_test.exs

```diff
@@ -0,0 +1,210 @@
+defmodule CollectorTest do
+  @moduledoc """
+  Define tests for the Query Collector Module
+  """
+
+  use ExUnit.Case, async: true
+  import Mox
+  alias Station.QueryCollector, as: QC
+  alias Util.Itinerary, as: Itinerary
+  alias Util.Query, as: Query
+  alias Util.Connection, as: Connection
+  alias Util.Preference, as: Preference
+
+  @dependency %{
+    uqc: MockUQC,
+    itinerary: MockItinerary,
+    registry: MockRegister,
+    station: MockStation
+  }
+
+  test "Normal working of Query Collector with number of completed itineraries equal to max_itineraries" do
+    query = %Query{
+      qid: "0300",
+      src_station: 0,
+      dst_station: 3,
+      arrival_time: 0,
+      end_time: 999_999
+    }
+
+    preference = %Preference{day: 0}
+
+    itinerary = Itinerary.new(query, preference)
+
+    # Set options for QC process
+    opts = %{max_itineraries: 5, timeout: 100}
+
+    dependency = @dependency
+
+    # Get 5 random terminal itineraries of length 4
+    itinerary1 = get_terminal_itinerary(query, preference, 4)
```

Comment:
reformat query to contain query_id and itinerary.

Reply:
Rename the existing itinerary variable to `itinerary_acc = {query, itinerary, preference}`. Please make this change in all the modules.